src/libcrun/handlers/spin.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * crun - OCI runtime written in C
+ *
+ * Copyright (C) 2017, 2018, 2019, 2020, 2021 Giuseppe Scrivano <giuseppe@scrivano.org>
+ * crun is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * crun is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with crun.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#define _GNU_SOURCE
+
+#include <config.h>
+#include "../custom-handler.h"
+#include "../linux.h"
+#include <sys/stat.h>
+
+#ifdef HAVE_DLOPEN
+#  include <dlfcn.h>
+#endif
+
+#ifdef HAVE_SPIN
+
+#endif
+
+#if HAVE_DLOPEN && HAVE_SPIN
+static int
+spin_exec (void *cookie arg_unused, libcrun_container_t *container arg_unused,
+           const char *pathname arg_unused, char *const argv[] arg_unused)
+{
+  // wasmtime fails to determine default config path if $HOME is not set
+  char *newenviron[] = { "HOME=/root", NULL };
+  char *newargv[] = { "/bin/spin", "up", "--listen", "--listen", NULL };
+
+  // spin up needs a /tmp folder
+  struct stat st = { 0 };
+  if (stat ("/tmp", &st) == -1)
+    {
+      mkdir ("/tmp", 0700);
```

Comment:
Should we handle error here, I think error handling must be similar to how its handled in other handlers for eg https://github.com/containers/crun/blob/main/src/libcrun/handlers/wasmer.c#L150

Reply:
If inside the container no /tmp folder is present, it should be created. Thats the case for almost all Spin images.
I'd not handle it as an error.

Reply:
No i mean the return value from `mkdir` is not handled.

Reply:
Oh, yes. That's an error that should be handled 😅

Reply:
Error handling added. 😊